dhall/src/Dhall/TypeCheck.hs

```diff
@@ -803,7 +805,7 @@ infer typer = loop
                     Dhall.Map.unorderedTraverseWithKey_ process (Dhall.Map.delete x₀ xTs)
 
                     return (VConst c₀)
-        Combine l r -> do
+        Combine mt l r -> do
```

Comment:
Nit: I believe we've used `mt` often for optional type fields. In this case I think `mk` could be less confusing.